static/js/ui_report.ts

```diff
@@ -19,16 +24,21 @@ export function message(response_html, status_box, cls = "alert", remove_after =
         .html(response_html)
         .stop(true)
         .fadeTo(0, 1);
-    if (remove_after) {
+    if (remove_after !== undefined) {
         setTimeout(() => {
             status_box.fadeOut(400);
         }, remove_after);
     }
     status_box.addClass("show");
 }
 
-export function error(response_html, xhr, status_box, remove_after) {
-    if (xhr && xhr.status.toString().charAt(0) === "4") {
+export function error(
+    response_html: string,
+    xhr: JQuery.jqXHR | undefined,
+    status_box: JQuery,
+    remove_after?: number,
+): void {
+    if (xhr && xhr.status.toString().startsWith("4")) {
```

Comment:
If we’re fixing this, we should just do it numerically: `xhr && xhr.status >= 400 && xhr.status < 500`.

Reply:
Updated in the prep commit.